File: 1030.距离顺序排列矩阵单元格/1030-距离顺序排列矩阵单元格.cpp
class Solution {
public:
    vector<vector<int>> allCellsDistOrder(int R, int C, int r0, int c0) {
        vector<vector<int>> ans;
        for( int r=0; r<R; r++ ) {
            for( int c=0; c<C; c++ ) {
                ans.push_back( {r, c} );
            }
        }

        sort( ans.begin(), ans.end(), [=](vector<int>& a, vector<int>& b)  {
            return (abs( a[0]-r0 ) + abs( a[1] - c0 )) < (abs( b[0]-r0 ) + abs( b[1] - c0 ));
        }  );

        return ans;
    }
};

File: 1030.距离顺序排列矩阵单元格/1030-距离顺序排列矩阵单元格.rs
impl Solution {
    pub fn all_cells_dist_order(r: i32, c: i32, r0: i32, c0: i32) -> Vec<Vec<i32>> {
        let mut ans : Vec<Vec<i32>> = Vec::new();
        for i in 0..r {
            for j in 0..c {
                ans.push( vec![i, j] );
            }
        }        

        ans.sort_by(|a, b| ( (a[0]-r0).abs() + (a[1]-c0).abs() ).cmp( &((b[0]-r0).abs() + (b[1]-c0).abs() )) ) ;

        return ans;
    }
}